src/test/java/mate/academy/onlinebookstore/service/BookServiceTest.java

```diff
@@ -0,0 +1,229 @@
+
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    @DisplayName("""
+    Should find Book by ID and return BookDto
+            """)
+    void findById_ValidId_ReturnsBookDto() {
+        BookDto expected = bookDto;
+
+        Mockito.when(bookRepository.findById(1L)).thenReturn(Optional.of(book));
+        Mockito.when(bookMapper.toDto(book)).thenReturn(expected);
+
+        BookDto actual = bookService.findById(1L);
+
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    @DisplayName("""
+    Should throw EntityNotFoundException when Book not found by ID
+            """)
+    void findById_InvalidId_ThrowsEntityNotFoundException() {
+        Mockito.when(bookRepository.findById(1L)).thenReturn(Optional.empty());
+
+        assertThrows(EntityNotFoundException.class, () -> bookService.findById(1L));
+    }
+
+    @Test
+    @DisplayName("Should update and return BookDto")
+    void updateById_ValidId_ReturnsUpdatedBookDto() {
+        UpdateBookRequestDto updateRequestDto = new UpdateBookRequestDto();
+        updateRequestDto.setTitle("new title");
+        updateRequestDto.setAuthor("new author");
+        updateRequestDto.setIsbn("2781234567897");
+        updateRequestDto.setPrice(BigDecimal.valueOf(29.99));
+        updateRequestDto.setDescription("new description");
+        updateRequestDto.setCoverImage("new cover image");
+        updateRequestDto.setCategoryIds(Set.of(2L));
+
+        Book updatedBook = book;
+        updatedBook.setTitle(updateRequestDto.getTitle());
+        updatedBook.setAuthor(updateRequestDto.getAuthor());
+        updatedBook.setIsbn(updateRequestDto.getIsbn());
+        updatedBook.setPrice(updateRequestDto.getPrice());
+        updatedBook.setDescription(updateRequestDto.getDescription());
+        updatedBook.setCoverImage(updateRequestDto.getCoverImage());
+        updatedBook.setCategories(Set.of(new Category(2L)));
+
+        BookDto expected = new BookDto();
+        expected.setId(1L);
+        updateRequestDto.setTitle("new title");
+        updateRequestDto.setAuthor("new author");
+        updateRequestDto.setIsbn("2781234567897");
+        updateRequestDto.setPrice(BigDecimal.valueOf(29.99));
+        updateRequestDto.setDescription("new description");
+        updateRequestDto.setCoverImage("new cover image");
+        updateRequestDto.setCategoryIds(Set.of(2L));
+
+        Mockito.when(bookRepository.findById(1L)).thenReturn(Optional.of(book));
+        Mockito.when(bookMapper.toDto(updatedBook)).thenReturn(expected);
+        Mockito.when(bookRepository.save(updatedBook)).thenReturn(updatedBook);
+
+        BookDto actual = bookService.updateById(1L, updateRequestDto);
+
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    @DisplayName("Should delete Book by ID")
+    void deleteById_ValidId_DeletesBook() {
+        bookService.deleteById(1L);
+        Mockito.verify(bookRepository).deleteById(1L);
```

Comment:
You have different implementations of `deleteById()` test method in `BookServiceTest` and `CategoryServiceTest` classes. Please, mock bookRepository call, as you did in `CategoryServiceTest`